src/deadline/job_attachments/bucket_sweeper/sweeper_processor.py

```
@@ -0,0 +1,200 @@
+            manifest_path (str): Local path to the manifest file
+            object_key (str): S3 object key for the uploaded manifest
+
+        Raises:
+            JobAttachmentS3BotoCoreError: If any errors occur during the upload process
+        """
+        try:
+            self.s3.upload_file(manifest_path, self.bucket_name, object_key)
+        except BotoCoreError as e:
+            raise JobAttachmentS3BotoCoreError(
+                action="uploading bucket sweeper tag manifest", error_details=str(e)
+            )
+
+    def _create_batch_tag_s3_job(self, s3_manifest_key: str) -> None:
+        """
+        Creates an S3 Batch Operations job to tag objects for deletion.
+
+        Args:
+            s3_manifest_key (str): Object key of the manifest file in S3
+
+        Raises:
+            JobAttachmentS3BotoCoreError: When retrieving manifest metadata fails
+            SweeperProcessorError: When creating the batch job fails
+        """
+        manifest_etag: str = self._get_manifest_etag(s3_manifest_key)
+        manifest: Dict[str, Any] = self._create_manifest_config(s3_manifest_key, manifest_etag)
+        operation: Dict[str, Any] = self._create_delete_tagging_operation()
+
+        self._submit_tagging_batch_job(
+            operation=operation,
+            manifest=manifest,
+        )
+
+    def _get_manifest_etag(self, s3_manifest_key: str) -> str:
+        """
+        Retrieves the ETag for the manifest file from S3.
+
+        Raises:
+            ValueError: when head_object operation succeeds but etag is None
+            JobAttachmentS3BotoCoreError: When S3 head_object operation fails
+        """
+        try:
+            manifest_metadata: Dict[str, Any] = self.s3.head_object(
+                Bucket=self.bucket_name, Key=s3_manifest_key
+            )
+
+            etag: str = manifest_metadata.get("ETag", "")
+            if not etag:
+                raise ValueError("Missing required Etag in manifest metadata")
```

Why throw an error here instead of returning `None`?

I think it's cleaner to return `None` for when `ETag` is None rather than have the caller catch `ValueError` (which can be thrown for reasons other than ETag being None)

---

We need the Etag to successfully create the S3 batch operations job. The batch operations will fail otherwise.

---

I will raise a JobAttachmentsSweeperError instead.